app/actions/local/team.test.ts

```diff
@@ -0,0 +1,390 @@
+        (operator.handleTeamSearchHistory as jest.Mock).mockResolvedValue([searchModel]);
+
+        const result = await addSearchToTeamSearchHistory(serverUrl, teamId, terms);
+
+        expect(operator.handleTeamSearchHistory).toHaveBeenCalledWith({
+            teamSearchHistories: [newSearch],
+            prepareRecordsOnly: true,
+        });
+        expect(operator.batchRecords).toHaveBeenCalledWith([searchModel], 'addSearchToTeamHistory');
+        expect(result).toEqual({searchModel});
+    });
+
+    it('should delete the oldest entry if team search history exceeds the limit', async () => {
+        const newSearch = {
+            created_at: Date.now(),
+            display_term: terms,
+            term: terms,
+            team_id: teamId,
+        };
+        const searchModel = {id: 'searchModelId', _raw: {_changed: ''}};
+        const oldSearchModel = {id: 'oldSearchModelId', prepareDestroyPermanently: jest.fn().mockReturnValue(undefined)};
+        const teamSearchHistory = new Array(MAX_TEAM_SEARCHES + 1).fill(oldSearchModel);
+
+        (operator.handleTeamSearchHistory as jest.Mock).mockResolvedValue([searchModel]);
+        (queryTeamSearchHistoryByTeamId as jest.Mock).mockReturnValue({
+            fetch: jest.fn().mockResolvedValue(teamSearchHistory),
+        });
+
+        const result = await addSearchToTeamSearchHistory(serverUrl, teamId, terms);
+
+        expect(operator.handleTeamSearchHistory).toHaveBeenCalledWith({
+            teamSearchHistories: [newSearch],
+            prepareRecordsOnly: true,
+        });
+        expect(queryTeamSearchHistoryByTeamId).toHaveBeenCalledWith(database, teamId);
+        expect(oldSearchModel.prepareDestroyPermanently).toHaveBeenCalledTimes(1);
+        expect(operator.batchRecords).toHaveBeenCalledWith([searchModel], 'addSearchToTeamHistory');
+        expect(result).toEqual({searchModel});
+    });
+
+    it('should handle case when searchModel._raw._changed is "created_at"', async () => {
+        const newSearch = {
+            created_at: Date.now(),
+            display_term: terms,
+            term: terms,
+            team_id: teamId,
+        };
+        const searchModel = {id: 'searchModelId', _raw: {_changed: 'created_at'}};
+        (operator.handleTeamSearchHistory as jest.Mock).mockResolvedValue([searchModel]);
+
+        const result = await addSearchToTeamSearchHistory(serverUrl, teamId, terms);
+
+        expect(operator.handleTeamSearchHistory).toHaveBeenCalledWith({
+            teamSearchHistories: [newSearch],
+            prepareRecordsOnly: true,
+        });
+        expect(queryTeamSearchHistoryByTeamId).not.toHaveBeenCalled();
+        expect(operator.batchRecords).toHaveBeenCalledWith([searchModel], 'addSearchToTeamHistory');
+        expect(result).toEqual({searchModel});
+    });
+
+    it('should handle case when teamSearchHistory length is less than or equal to MAX_TEAM_SEARCHES', async () => {
```

Comment:
Maybe we should check for prepareDestroyPermanently() to not have been called here?

Reply:
done.